CMakeLists.txt

```diff
@@ -34,9 +34,12 @@ if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
                                                "MinSizeRel" "RelWithDebInfo")
 endif()
 
+link_directories("/usr/local/lib")
 if(USE_PNG)
   find_package(ZLIB REQUIRED)
-  find_package(PNG REQUIRED)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(PNG libpng16 REQUIRED)
+#  find_package(PNG REQUIRED)
```

Comment:
Why was this necessary? Are there some issues with find_package?

Reply:
In order to get the PNG build to work on the Mac.

Reply:
Then the changes required to fix broken PNG installation on Mac should be in:
```
if(APPLE)
 ...
endif()
```

Reply:
I had no issues with `find_package` on my macOS linking w/ PNG.
I can still build with this:
```
#link_directories("/usr/local/lib")
if(USE_PNG)
  find_package(ZLIB REQUIRED)
  #find_package(PkgConfig REQUIRED)
  #pkg_check_modules(PNG libpng16 REQUIRED)
  find_package(PNG REQUIRED)
else()
  message(STATUS "Will not build PNG support")
endif()
```
```
The library builds and all test pass.

Reply:
The issue is not really w/ macOS.  Its got to do w/ how the Github Action macOS is configured.
`brew` is used to install `libpng`, which is likely the issue that causes the action to fail.
Like, `jasper`, if PNG is built and cached, I am sure this won't be an issue and we can revert to "standard operating procedures"
I don't have time to investigate further at the moment.